source/colors.py

```diff
@@ -1,24 +1,62 @@
-#colors.py
-#A part of NonVisual Desktop Access (NVDA)
-#Copyright (C) 2006-2008 NVDA Contributors <http://www.nvda-project.org/>
-#This file is covered by the GNU General Public License.
-#See the file COPYING for more details.
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2006-2021 NV Access Limited, Babbage B.V., Joseph Lee, Mesar Hameed, kvark128
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
 
 from collections import namedtuple
 import math
 import colorsys
 from ctypes.wintypes import COLORREF
 import re
+from functools import lru_cache
+from typing import Union
+
+#: Flag to indicate color being decoded from displayModelFormatColor_t
+# is transparent.
+# See (displayModel.cpp in nvdaHelper) displayModelFormatColor_t::TRANSPARENT_BIT
+TRANSPARENT_BITFLAG = 0x01 << 24
 
 class RGB(namedtuple('RGB',('red','green','blue'))):

```

Comment:
I wonder whether we should consider changing this into a dataclass in 2022.1

Reply:
Yes, or given it's complexity a regular class might be more appropriate.

Reply:
I won't include this change here, I think this deserves its own conversation. Happy to see a PR if you would like to experiment with it.